### PR TITLE
feat(medusa): remove created reservations on subsequent failure for cart completion

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- If cart completion fails after creating reservations, remove those reservations

**Why**
- To avoid hanging reservations if something fails at a later point

### CODE REVIEW

**Review by srindom — CHANGES_REQUESTED** [review]
Please add an integration test.

**Comment by olivermrbl** [comment]
@srindom Will you give this a second review? 🙏

**Review by srindom — APPROVED** [review]
LGTM w. comments.

File: integration-tests/plugins/__tests__/inventory/cart/cart.js [hunk 1]
```diff
@@ -191,6 +191,47 @@ describe("/store/carts", () => {
...
+      const getRes = await api
+        .post(`/store/carts/${cartId}/complete`)
+        .catch((err) => err)
+
+      expect(getRes.response.status).toEqual(500)
+      expect(getRes.response.data).toEqual({
+        code: "unknown_error",
+        type: "unknown_error",
+        message: "An unknown error occurred.",
+      })
```
[the remark this answers is not in the document]
Comment (pKorsholm) [reply integration-tests/plugins/__tests__/inventory/cart/cart.js]: Added it here 😄 it was just adding a case to the exception formatter, but we should do some extensive exploration into different postgres error codes. This was cause by trying to complete an order without a customer which yielded no clear error message, however it should 😄

**Pull request merged**